### ISSUE TITLE
explicit ecmaVersion of `5` is reported as an invalid ecmaVersion

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#12101]
**Tell us about your environment**

* **ESLint Version:** v6.1.0
* **Node Version:** v12.8.0
* **npm Version:** v6.10.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>.eslintrc</summary>

```js
{
	"root": true,

	"extends": "@ljharb",

	"parserOptions": {
		"ecmaVersion": "5",
	},

	"rules": {
		"comma-dangle": [2, "always-multiline"],
		"func-name-matching": [2, "always"],
		"dot-notation": [2, { "allowKeywords": true }],
		"max-statements": 0,
	},
}
```
</details>

<details>
<summary>--print-config Configuration</summary>

```js
{
  "env": {
    "es6": false,
    "browser": false,
    "node": true,
    "amd": false,
    "mocha": false,
    "jasmine": false
  },
  "globals": {
    "BigInt": false,
    "Symbol": false,
    "Map": false,
    "Set": false,
    "Reflect": false
  },
  "parser": null,
  "parserOptions": {
    "ecmaVersion": "5",
    "sourceType": "script"
  },
  "plugins": [],
  "rules": {
    "comma-dangle": [
      2,
      "always-multiline"
    ],
    "func-name-matching": [
      2,
      "always"
    ],
    "dot-notation": [
      2,
      {
        "allowKeywords": true,
        "allowPattern": ""
      }
    ],
    "max-statements": [
      0,
      10
    ],
    "accessor-pairs": [
      2,
      {
        "getWithoutSet": false,
        "setWithoutGet": true
      }
    ],
    "array-bracket-newline": [
      2,
      {
        "multiline": true,
        "minItems": 3
      }
    ],
    "array-bracket-spacing": [
      2,
      "never",
      {
        "singleValue": false,
        "objectsInArrays": false,
        "arraysInArrays": false
      }
    ],
    "array-callback-return": [
      2
    ],
    "array-element-newline": [
      2,
      "consistent"
    ],
    "arrow-body-style": [
      2,
      "as-needed"
    ],
    "arrow-parens": [
      2,
      "as-needed",
      {
        "requireForBlockBody": false
      }
    ],
    "arrow-spacing": [
      2,
      {
        "before": true,
        "after": true
      }
    ],
    "block-scoped-var": [
      0
    ],
    "block-spacing": [
      2,
      "always"
    ],
    "brace-style": [
      2,
      "1tbs",
      {
        "allowSingleLine": true
      }
    ],
    "callback-return": [
      2,
      [
        "callback"
      ]
    ],
    "camelcase": [
      2,
      {
        "properties": "always",
        "ignoreDestructuring": false,
        "allow": []
      }
    ],
    "capitalized-comments": [
      0
    ],
    "class-methods-use-this": [
      2,
      {
        "exceptMethods": []
      }
    ],
    "comma-spacing": [
      2,
      {
        "before": false,
        "after": true
      }
    ],
    "comma-style": [
      2,
      "last"
    ],
    "complexity": [
      2,
      10
    ],
    "computed-property-spacing": [
      2,
      "never"
    ],
    "consistent-return": [
      2
    ],
    "consistent-this": [
      0,
      "that"
    ],
    "constructor-super": [
      2
    ],
    "curly": [
      2,
      "all"
    ],
    "default-case": [
      2,
      {
        "commentPattern": "(?:)"
      }
    ],
    "dot-location": [
      2,
      "property"
    ],
    "eol-last": [
      2,
      "always"
    ],
    "eqeqeq": [
      2
    ],
    "for-direction": [
      2
    ],
    "func-call-spacing": [
      2,
      "never"
    ],
    "func-names": [
      0,
      "always",
      {
        "generators": "always"
      }
    ],
    "func-style": [
      2,
      "expression"
    ],
    "function-paren-newline": [
      2,
      "multiline"
    ],
    "generator-star-spacing": [
      2,
      {
        "before": false,
        "after": true,
        "anonymous": {
          "before": false,
          "after": true
        },
        "method": {
          "before": false,
          "after": true
        },
        "named": {
          "before": false,
          "after": true
        }
      }
    ],
    "getter-return": [
      2,
      {
        "allowImplicit": true
      }
    ],
    "global-require": [
      2
    ],
    "global-strict": [
      0,
      "never"
    ],
    "guard-for-in": [
      0
    ],
    "handle-callback-err": [
      0
    ],
    "id-blacklist": [
      0
    ],
    "id-length": [
      2,
      {
        "min": 1,
        "max": 20,
        "properties": "never"
      }
    ],
    "id-match": [
      0
    ],
    "implicit-arrow-linebreak": [
      2,
      "beside"
    ],
    "indent": [
      2,
      "tab",
      {
        "SwitchCase": 1,
        "VariableDeclarator": 1,
        "outerIIFEBody": 1,
        "FunctionDeclaration": {
          "parameters": 1,
          "body": 1
        },
        "FunctionExpression": {
          "parameters": 1,
          "body": 1
        },
        "CallExpression": {
          "arguments": 1
        },
        "ArrayExpression": 1,
        "ObjectExpression": 1,
        "ImportDeclaration": 1,
        "flatTernaryExpressions": false,
        "ignoredNodes": [
          "JSXElement *"
        ],
        "ignoreComments": false
      }
    ],
    "init-declarations": [
      0,
      "always"
    ],
    "jsx-quotes": [
      2,
      "prefer-double"
    ],
    "key-spacing": [
      2,
      {
        "beforeColon": false,
        "afterColon": true,
        "mode": "strict"
      }
    ],
    "keyword-spacing": [
      2,
      {
        "before": true,
        "after": true,
        "overrides": {
          "return": {
            "after": true,
            "before": true
          },
          "throw": {
            "after": true,
            "before": true
          },
          "case": {
            "after": true,
            "before": true
          }
        }
      }
    ],
    "linebreak-style": [
      2,
      "unix"
    ],
    "line-comment-position": [
      0,
      {
        "position": "beside",
        "ignorePattern": "",
        "applyDefaultPatterns": true
      }
    ],
    "lines-around-comment": [
      2,
      {
        "beforeBlockComment": false,
        "afterBlockComment": false,
        "beforeLineComment": false,
        "allowBlockStart": true,
        "allowBlockEnd": true,
        "afterLineComment": false
      }
    ],
    "lines-around-directive": [
      2,
      {
        "before": "always",
        "after": "always"
      }
    ],
    "lines-between-class-members": [
      2,
      "always",
      {
        "exceptAfterSingleLine": false
      }
    ],
    "max-classes-per-file": [
      0,
      1
    ],
    "max-depth": [
      2,
      4
    ],
    "max-len": [
      2,
      120,
      4,
      {
        "ignoreUrls": true,
        "ignoreComments": true,
        "ignoreRegExpLiterals": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true
      }
    ],
    "max-lines-per-function": [
      2,
      {
        "max": 50,
        "skipBlankLines": true,
        "skipComments": true,
        "IIFEs": true
      }
    ],
    "max-lines": [
      2,
      {
        "max": 300,
        "skipBlankLines": true,
        "skipComments": true
      }
    ],
    "max-nested-callbacks": [
      2,
      2
    ],
    "max-params": [
      2,
      2
    ],
    "max-statements-per-line": [
      2,
      {
        "max": 1
      }
    ],
    "multiline-comment-style": [
      2,
      "starred-block"
    ],
    "multiline-ternary": [
      0,
      "always-multiline"
    ],
    "new-parens": [
      2
    ],
    "new-cap": [
      2,
      {
        "capIsNewExceptions": [
          "Symbol",
          "BigInt"
        ],
        "newIsCap": true,
        "capIsNew": true,
        "properties": true
      }
    ],
    "newline-after-var": [
      0
    ],
    "newline-before-return": [
      0
    ],
    "newline-per-chained-call": [
      2,
      {
        "ignoreChainWithDepth": 6
      }
    ],
    "no-alert": [
      2
    ],
    "no-array-constructor": [
      2
    ],
    "no-async-promise-executor": [
      2
    ],
    "no-await-in-loop": [
      2
    ],
    "no-compare-neg-zero": [
      2
    ],
    "no-confusing-arrow": [
      2
    ],
    "no-bitwise": [
      0
    ],
    "no-buffer-constructor": [
      2
    ],
    "no-caller": [
      2
    ],
    "no-case-declarations": [
      2
    ],
    "no-catch-shadow": [
      2
    ],
    "no-class-assign": [
      2
    ],
    "no-cond-assign": [
      2
    ],
    "no-console": [
      2
    ],
    "no-const-assign": [
      2
    ],
    "no-constant-condition": [
      2
    ],
    "no-continue": [
      2
    ],
    "no-control-regex": [
      2
    ],
    "no-debugger": [
      2
    ],
    "no-delete-var": [
      2
    ],
    "no-div-regex": [
      0
    ],
    "no-dupe-args": [
      2
    ],
    "no-dupe-class-members": [
      2
    ],
    "no-dupe-keys": [
      2
    ],
    "no-duplicate-case": [
      2
    ],
    "no-duplicate-imports": [
      2
    ],
    "no-else-return": [
      0
    ],
    "no-empty": [
      2,
      {
        "allowEmptyCatch": true
      }
    ],
    "no-empty-character-class": [
      2
    ],
    "no-empty-function": [
      2,
      {
        "allow": [
          "arrowFunctions",
          "functions"
        ]
      }
    ],
    "no-empty-pattern": [
      2
    ],
    "no-eq-null": [
      0
    ],
    "no-eval": [
      2
    ],
    "no-ex-assign": [
      2
    ],
    "no-extend-native": [
      2
    ],
    "no-extra-bind": [
      2
    ],
    "no-extra-boolean-cast": [
      2
    ],
    "no-extra-label": [
      2
    ],
    "no-extra-parens": [
      2,
      "all",
      {
        "conditionalAssign": false,
        "nestedBinaryExpressions": false,
        "returnAssign": false,
        "ignoreJSX": "all",
        "enforceForArrowConditionals": false
      }
    ],
    "no-extra-semi": [
      2
    ],
    "no-fallthrough": [
      2
    ],
    "no-floating-decimal": [
      2
    ],
    "no-func-assign": [
      2
    ],
    "no-implied-eval": [
      2
    ],
    "no-implicit-coercion": [
      2,
      {
        "boolean": false,
        "number": true,
        "string": true
      }
    ],
    "no-implicit-globals": [
      2
    ],
    "no-inline-comments": [
      0
    ],
    "no-inner-declarations": [
      2,
      "functions"
    ],
    "no-invalid-regexp": [
      2
    ],
    "no-invalid-this": [
      2
    ],
    "no-irregular-whitespace": [
      2
    ],
    "no-iterator": [
      2
    ],
    "no-label-var": [
      2
    ],
    "no-labels": [
      2,
      {
        "allowLoop": false,
        "allowSwitch": false
      }
    ],
    "no-lone-blocks": [
      2
    ],
    "no-lonely-if": [
      2
    ],
    "no-loop-func": [
      2
    ],
    "no-magic-numbers": [
      1,
      {
        "ignore": [],
        "enforceConst": false,
        "detectObjects": true,
        "ignoreArrayIndexes": false
      }
    ],
    "no-misleading-character-class": [
      2
    ],
    "no-mixed-operators": [
      2,
      {
        "groups": [
          [
            "+",
            "-",
            "*",
            "/",
            "%",
            "**"
          ],
          [
            "&",
            "|",
            "^",
            "~",
            "<<",
            ">>",
            ">>>"
          ],
          [
            "==",
            "!=",
            "===",
            "!==",
            ">",
            ">=",
            "<",
            "<="
          ],
          [
            "&&",
            "||"
          ],
          [
            "in",
            "instanceof"
          ]
        ],
        "allowSamePrecedence": true
      }
    ],
    "no-mixed-requires": [
      0,
      false
    ],
    "no-mixed-spaces-and-tabs": [
      2,
      false
    ],
    "no-multi-assign": [
      2
    ],
    "no-multi-spaces": [
      2,
      {
        "ignoreEOLComments": false
      }
    ],
    "no-multi-str": [
      2
    ],
    "no-multiple-empty-lines": [
      2,
      {
        "max": 1,
        "maxEOF": 1
      }
    ],
    "no-global-assign": [
      2,
      {
        "exceptions": []
      }
    ],
    "no-native-reassign": [
      2
    ],
    "no-negated-condition": [
      2
    ],
    "no-negated-in-lhs": [
      2
    ],
    "no-nested-ternary": [
      0
    ],
    "no-new": [
      2
    ],
    "no-new-func": [
      2
    ],
    "no-new-object": [
      2
    ],
    "no-new-require": [
      0
    ],
    "no-new-symbol": [
      2
    ],
    "no-new-wrappers": [
      2
    ],
    "no-obj-calls": [
      2
    ],
    "no-octal": [
      2
    ],
    "no-octal-escape": [
      2
    ],
    "no-param-reassign": [
      2
    ],
    "no-path-concat": [
      0
    ],
    "no-plusplus": [
      2,
      {
        "allowForLoopAfterthoughts": true
      }
    ],
    "no-process-env": [
      0
    ],
    "no-process-exit": [
      2
    ],
    "no-proto": [
      2
    ],
    "no-prototype-builtins": [
      2
    ],
    "no-redeclare": [
      2
    ],
    "no-regex-spaces": [
      2
    ],
    "no-restricted-globals": [
      0
    ],
    "no-restricted-imports": [
      0
    ],
    "no-restricted-modules": [
      0
    ],
    "no-restricted-properties": [
      2,
      {
        "object": "Symbol",
        "property": "for",
        "message": "Symbol.for is unshimmable"
      },
      {
        "object": "arguments",
        "property": "callee",
        "message": "arguments.callee is deprecated"
      },
      {
        "property": "__defineGetter__",
        "message": "Please use Object.defineProperty instead."
      },
      {
        "property": "__defineSetter__",
        "message": "Please use Object.defineProperty instead."
      }
    ],
    "no-restricted-syntax": [
      2,
      "BreakStatement",
      "ContinueStatement",
      "DebuggerStatement",
      "ForInStatement",
      "LabeledStatement",
      "WithStatement"
    ],
    "no-return-assign": [
      2,
      "always"
    ],
    "no-return-await": [
      2
    ],
    "no-script-url": [
      2
    ],
    "no-self-assign": [
      2,
      {
        "props": true
      }
    ],
    "no-self-compare": [
      0
    ],
    "no-sequences": [
      2
    ],
    "no-shadow": [
      2,
      {
        "builtinGlobals": false,
        "hoist": "functions",
        "allow": []
      }
    ],
    "no-shadow-restricted-names": [
      2
    ],
    "no-spaced-func": [
      2
    ],
    "no-sparse-arrays": [
      2
    ],
    "no-sync": [
      0
    ],
    "no-tabs": [
      0
    ],
    "no-template-curly-in-string": [
      2
    ],
    "no-ternary": [
      0
    ],
    "no-this-before-super": [
      2
    ],
    "no-throw-literal": [
      2
    ],
    "no-trailing-spaces": [
      2,
      {
        "skipBlankLines": false,
        "ignoreComments": false
      }
    ],
    "no-undef": [
      2,
      {
        "typeof": true
      }
    ],
    "no-undef-init": [
      2
    ],
    "no-undefined": [
      0
    ],
    "no-underscore-dangle": [
      2,
      {
        "allow": [],
        "allowAfterThis": false,
        "allowAfterSuper": false,
        "enforceInMethodNames": true
      }
    ],
    "no-unexpected-multiline": [
      2
    ],
    "no-unmodified-loop-condition": [
      2
    ],
    "no-unneeded-ternary": [
      2
    ],
    "no-unreachable": [
      2
    ],
    "no-unsafe-finally": [
      2
    ],
    "no-unsafe-negation": [
      2
    ],
    "no-unused-expressions": [
      2,
      {
        "allowShortCircuit": false,
        "allowTernary": false,
        "allowTaggedTemplates": true
      }
    ],
    "no-unused-labels": [
      2
    ],
    "no-unused-vars": [
      2,
      {
        "vars": "all",
        "args": "after-used"
      }
    ],
    "no-use-before-define": [
      2
    ],
    "no-useless-call": [
      2
    ],
    "no-useless-catch": [
      2
    ],
    "no-useless-computed-key": [
      2
    ],
    "no-useless-concat": [
      2
    ],
    "no-useless-constructor": [
      2
    ],
    "no-useless-escape": [
      2
    ],
    "no-useless-rename": [
      2,
      {
        "ignoreDestructuring": false,
        "ignoreImport": false,
        "ignoreExport": false
      }
    ],
    "no-useless-return": [
      2
    ],
    "no-var": [
      0
    ],
    "no-void": [
      0
    ],
    "no-warning-comments": [
      0,
      {
        "terms": [
          "todo",
          "fixme",
          "xxx"
        ],
        "location": "start"
      }
    ],
    "no-whitespace-before-property": [
      2
    ],
    "no-with": [
      2
    ],
    "nonblock-statement-body-position": [
      2,
      "beside"
    ],
    "object-curly-newline": [
      2,
      {
        "ObjectExpression": {
          "multiline": true,
          "consistent": true,
          "minProperties": 3
        },
        "ObjectPattern": {
          "multiline": true,
          "consistent": true,
          "minProperties": 3
        }
      }
    ],
    "object-curly-spacing": [
      2,
      "always"
    ],
    "object-property-newline": [
      0,
      {
        "allowMultiplePropertiesPerLine": true
      }
    ],
    "object-shorthand": [
      2,
      "never"
    ],
    "one-var": [
      0
    ],
    "one-var-declaration-per-line": [
      2,
      "initializations"
    ],
    "operator-assignment": [
      0,
      "always"
    ],
    "operator-linebreak": [
      2,
      "none"
    ],
    "padded-blocks": [
      0
    ],
    "padding-line-between-statements": [
      0
    ],
    "prefer-arrow-callback": [
      0,
      {
        "allowNamedFunctions": true,
        "allowUnboundThis": true
      }
    ],
    "prefer-const": [
      0,
      {
        "destructuring": "any",
        "ignoreReadBeforeAssign": true
      }
    ],
    "prefer-destructuring": [
      0,
      {
        "VariableDeclarator": {
          "array": false,
          "object": true
        },
        "AssignmentExpression": {
          "array": true,
          "object": true
        }
      },
      {
        "enforceForRenamedProperties": false
      }
    ],
    "prefer-named-capture-group": [
      0
    ],
    "prefer-numeric-literals": [
      0
    ],
    "prefer-object-spread": [
      0
    ],
    "prefer-promise-reject-errors": [
      2,
      {
        "allowEmptyReject": true
      }
    ],
    "prefer-reflect": [
      0
    ],
    "prefer-rest-params": [
      0
    ],
    "prefer-spread": [
      0
    ],
    "prefer-template": [
      0
    ],
    "quote-props": [
      2,
      "as-needed",
      {
        "keywords": true
      }
    ],
    "quotes": [
      2,
      "single",
      "avoid-escape"
    ],
    "radix": [
      2
    ],
    "require-atomic-updates": [
      2
    ],
    "require-await": [
      0
    ],
    "require-jsdoc": [
      0
    ],
    "require-unicode-regexp": [
      0
    ],
    "require-yield": [
      2
    ],
    "rest-spread-spacing": [
      2,
      "never"
    ],
    "semi": [
      2
    ],
    "semi-spacing": [
      2,
      {
        "before": false,
        "after": true
      }
    ],
    "semi-style": [
      2,
      "last"
    ],
    "sort-imports": [
      0
    ],
    "sort-keys": [
      2,
      "asc",
      {
        "caseSensitive": false,
        "natural": true,
        "minKeys": 2
      }
    ],
    "sort-vars": [
      0
    ],
    "space-before-blocks": [
      2,
      {
        "functions": "always",
        "keywords": "always",
        "classes": "always"
      }
    ],
    "space-before-function-paren": [
      2,
      {
        "anonymous": "always",
        "named": "never",
        "asyncArrow": "always"
      }
    ],
    "space-in-brackets": [
      0,
      "never",
      {
        "singleValue": true,
        "arraysInArrays": false,
        "arraysInObjects": false,
        "objectsInArrays": true,
        "objectsInObjects": true,
        "propertyName": false
      }
    ],
    "space-in-parens": [
      2,
      "never"
    ],
    "space-infix-ops": [
      2
    ],
    "space-unary-ops": [
      2,
      {
        "words": true,
        "nonwords": false
      }
    ],
    "spaced-comment": [
      2,
      "always",
      {
        "markers": [
          "!"
        ]
      }
    ],
    "spaced-line-comment": [
      0,
      "always"
    ],
    "strict": [
      2,
      "global"
    ],
    "switch-colon-spacing": [
      2,
      {
        "after": true,
        "before": false
      }
    ],
    "symbol-description": [
      2
    ],
    "template-curly-spacing": [
      2,
      "never"
    ],
    "template-tag-spacing": [
      2,
      "never"
    ],
    "unicode-bom": [
      2,
      "never"
    ],
    "use-isnan": [
      2
    ],
    "valid-jsdoc": [
      0
    ],
    "valid-typeof": [
      2
    ],
    "vars-on-top": [
      0
    ],
    "wrap-iife": [
      2,
      "outside",
      {
        "functionPrototypeMethods": true
      }
    ],
    "wrap-regex": [
      2
    ],
    "yield-star-spacing": [
      2,
      {
        "before": false,
        "after": true
      }
    ],
    "yoda": [
      2,
      "never",
      {
        "exceptRange": true,
        "onlyEquality": false
      }
    ]
  },
  "settings": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
'use strict';

var api = require('./api');

api.register();

module.exports = api;
```

```bash
eslint index.js
```

**What did you expect to happen?**
Linting.

**What actually happened? Please include the actual, raw output from ESLint.**
$PWD/index.js
  0:0  error  Parsing error: Invalid ecmaVersion

✖ 1 problem (1 error, 0 warnings)

**Are you willing to submit a pull request to fix this bug?**
I'm not sure what to fix.

**Comment by mysticatea** [comment]
Thank you for your report.

This is intentional behavior. The `ecmaVersion` option requires a number rather than strings.

**Comment by ljharb** [comment]
Thanks; can you help me find where in the changelog this breaking change is mentioned?

Also, perhaps the error message for the wrong type, could be different from the message for the wrong value?

**Comment by aladdin-add** [comment]
changelog: https://github.com/eslint/eslint/blob/master/docs/user-guide/migrating-to-6.0.0.md#-the-default-parser-now-validates-options-more-strictly

refs:  eslint/eslint#9687, eslint/espree#384

and I agreed the error message can be improved! :-)

**Comment by aladdin-add** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
